core/src/test/scala/akka/persistence/cassandra/EventsByTagRestartSpec.scala

```diff
@@ -83,5 +84,43 @@ class EventsByTagRestartSpec extends CassandraSpec(EventsByTagRestartSpec.config
       tagProbe.expectNoMessage(waitTime)
       tagProbe.cancel()
     }
+
+    "use correct tag sequence nrs when actor is stopped prematurely" in {
+      val queryJournal = PersistenceQuery(system).readJournalFor[CassandraReadJournal](CassandraReadJournal.Identifier)
+      val p2 = system.actorOf(TestTaggingActor.props("p2", Set("green")))
+      val probe1 = TestProbe()
+      probe1.watch(p2)
+      p2.tell("e1", probe1.ref)
+      probe1.expectMsg(Ack)
+      p2.tell("e2", probe1.ref)
+      probe1.expectMsg(Ack)
+      p2.tell("e3", probe1.ref)
+      probe1.expectMsg(Ack)
+      p2.tell("e4", probe1.ref)
+      // the PoisonPill will cause the actor to be stopped before the journal write has been completed,
+      // and is therefore similar to if the CircuitBreaker would trigger
+      p2 ! PoisonPill
+      probe1.expectTerminated(p2)
+
+      Thread.sleep(2000)
+      val probe2 = TestProbe()
+      // tag will be missing until we restart actor, (is that ok?)
```

Comment:
This is the big question with this solution, what if the actor is not restarted any time soon?

Reply:
This isn't a new limitation, just a new case, we always had the case if the app were to crash that inflight writes wouldn't be fixed until the persistent actor started again. 

This one is a shame as we know it has happened as we've dropped the write, but if we try to do something about it it could be interleaved by the actor restarting which gets tricky. 

That said, the `TagWriters` actor could start a fixing process and abandon it, or block the restart for the PID.

The `TagWriters` actor receives a PID starting before any writes can come for it as it has to ack the start message before recovery happens. This does mean that it could finish the fixing process.  

The tricky part is there could be the same PID running on a different node, say if an actor terminated prematurely and then was moved. That makes me think we should leave this as is. 

WDYT @johanandren @patriknw ?

Reply:
I think you are right, didn't think of it being the same as current behavior. Ok, that means this fix is an ok solution then.

Reply:
I think it would be good if we could avoid complicating it more than necessary. It's not great to have the journal event written without the corresponding tagged representation until the actor is started again (which could be never), but let's keep it simple as a first step and possibly add some "continue/repair" process separately (if requested) since there might be other cases when that could happen.